src/classes/partialsetaccumulator.cpp

```diff
@@ -0,0 +1,197 @@
+                            unboundPartials_[{i, j}].tag(), source.unboundPartial(i, j).tag())));
+        unboundPartials_[{i, j}] += source.unboundPartial(i, j);
+    });
+
+    total_ += source.total();
+
+    ++nAccumulated_;
+}
+
+/*
+ * Partials Data
+ */
+
+// Return number of accumulated points
+int PartialSetAccumulator::nAccumulated() const { return nAccumulated_; }
+
+// Return full matrix, containing sampling of full atom-atom partial
+const Array2D<SampledData1D> &PartialSetAccumulator::partials() const { return partials_; }
+
+// Return bound matrix, containing sampling of atom-atom partial of bound pairs
+const Array2D<SampledData1D> &PartialSetAccumulator::boundPartials() const { return boundPartials_; }
+
+// Return unbound matrix, containing sampling of atom-atom partial of unbound pairs
+const Array2D<SampledData1D> &PartialSetAccumulator::unboundPartials() const { return unboundPartials_; }
+
+// Return the sampled total function
+const SampledData1D &PartialSetAccumulator::total() const { return total_; }
+
+/*
+ * Searchers
+ */
+
+// Return SampledData1D with specified tag, if it exists
+OptionalReferenceWrapper<const SampledData1D> PartialSetAccumulator::searchSampledData1D(std::string_view tag) const
+{
+    auto fullIt = std::find_if(partials_.begin(), partials_.end(), [tag](const auto &data) { return data.tag() == tag; });
+    if (fullIt != partials_.end())
+        return *fullIt;
+    auto boundIt =
+        std::find_if(boundPartials_.begin(), boundPartials_.end(), [tag](const auto &data) { return data.tag() == tag; });
+    if (boundIt != boundPartials_.end())
+        return *boundIt;
+    auto unboundIt =
+        std::find_if(unboundPartials_.begin(), unboundPartials_.end(), [tag](const auto &data) { return data.tag() == tag; });
+    if (unboundIt != unboundPartials_.end())
+        return *unboundIt;
+    if (total_.tag() == tag)
+        return total_;
+    return {};
+}
+
+/*
+ * Serialisation
+ */
+
+// Read array through specified LineParser
+bool PartialSetAccumulator::deserialiseArray(LineParser &parser, Array2D<SampledData1D> &array,
+                                             const std::vector<double> &xAxis)
+{
+    for (auto &partial : array)
+    {
+        // Read tag
+        if (parser.readNextLine(LineParser::Defaults) != LineParser::Success)
+            return false;
+        partial.setTag(parser.line());
+        // Initialise from existing abscissa vector and read in sampled values
+        partial.initialise(xAxis);
+        if (!partial.deserialiseValues(parser))
+            return false;
+    }
+
+    return true;
+}
+
+// Write array through specified LineParser
+bool PartialSetAccumulator::serialiseArray(LineParser &parser, const Array2D<SampledData1D> &array)
+{
+    for (auto &partial : array)
+    {
+        if (!parser.writeLine(partial.tag()))
+            return false;
+        if (!partial.serialiseValues(parser))
+            return false;
+    }
+
+    return true;
```

Comment:
```suggestion
    return std::all_of(array.begin(), array.end(), [&parser](auto& partial){ return parser.writeLine(partial.tag()) && partial.serialiseValues(parser);});
```

Possible algorithm switch

Reply:
As a caution, we'll stick with the explicit loop as `std::any_of` potentially isn't guaranteed to respect ordering of items in the container.